src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,151 @@
+          break;
+        default:
+          break;
+      }
+    }
+
+    if (centuries.length) {
+      newList = newList.filter(person => centuries
+        .includes(`${Math.ceil(person.born / 100)}` as CenturiesOptions));
+    }
+
+    if (query) {
+      const normalizedQuery = query.toLowerCase();
+
+      newList = newList.filter(person => person.name.toLowerCase()
+        .includes(normalizedQuery)
+        || person.motherName?.toLowerCase()
+          .includes(normalizedQuery)
+        || person.fatherName?.toLowerCase()
+          .includes(normalizedQuery));
+    }
+
+    if (sort) {
+      newList.sort((firstPerson, secondPerson) => {
+        if (sort === SortOptions.NAME || sort === SortOptions.SEX) {
+          return (order ? -1 : 1)
+            * firstPerson[sort].localeCompare(secondPerson[sort]);
+        }
+
+        if (sort === SortOptions.BORN || sort === SortOptions.DIED) {
+          return (order ? -1 : 1)
+            * (firstPerson[sort] - secondPerson[sort]);
+        }
+
+        return 0;
+      });
+    }
+
+    setCurrentPeople(newList);
+  },
+  [people, searchParams]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const curQuery = event.target.value.trim()
+      ? event.target.value
+      : null;
+
+    setSearchParams(getSearchWith(searchParams, { query: curQuery }));
+  };
+
+  const tableValidate = !isLoading && !isError;
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
           <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
+            {!isLoading && !!people.length && !isError
+              && (
+                <PeopleFilters
+                  sex={sex}
+                  centuries={centuries}
+                  query={query}
+                  handleQueryChange={handleQueryChange}
+                />
+              )}
           </div>
 
           <div className="column">
             <div className="box table-container">
-              <Loader />
+              {isLoading && <Loader />}
 
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+              {isError && (
+                <p data-cy="peopleLoadingError">
+                  Something went wrong
+                </p>
+              )}
 
-              <p data-cy="noPeopleMessage">
-                There are no people on the server
-              </p>
+              {tableValidate && !people.length && (
+                <p data-cy="noPeopleMessage">
+                  There are no people on the server
+                </p>
+              )}
 
-              <p>There are no people matching the current search criteria</p>
+              {tableValidate && !currentPeople.length && !!people.length && (
+                <p data-cy="noPeopleMessage">
```

Comment:
The message 'There are no people on the server' is misleading. The message should be 'There are no people matching the current search criteria'.